src/shifter_core.c

```diff
@@ -3209,23 +3209,36 @@ int validateUnmounted(const char *path, int subtree) {
     return -1;
 }
 
-static char **_shifter_findenv(char ***env, const char *var, size_t n, size_t *nElement) {
+char **_shifter_findenv(char ***env, const char *var, size_t n, size_t *nElement) {
     char **ptr = NULL;
     char **ret = NULL;
+    char *key = NULL;
+    size_t len = 0;
     if (env == NULL || *env == NULL || var == NULL || n == 0) {
         return NULL;
     }
+
+    len = strlen(var);
+    if (len < n) {
+        len = n;
+    }
+    /* allocating an extra byte to be sure we can terminate */
+    key = _malloc(sizeof(char) * (len + 2));
+    strncpy(key, var, len + 2);
```

Comment:
Why copy +2 if you are just going to overwrite them in the next two lines?

Reply:
Disregard.  I understand now.  I guess if the search was var="ab", n=4 it would be safe but would fail.